backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java

```diff
@@ -245,10 +247,81 @@ public TestOrder editQueueItem(
 
+  private TestResult getCovidResultFromMultiplexList(List<MultiplexTestResult> results) {
+    return results.stream()
+        .filter(result -> result.getDiseaseName().equals("COVID-19"))
+        .findFirst()
+        .orElseThrow(() -> new IllegalArgumentException("No COVID-19 test result found"))
+        .getTestResult();
+  }
+
+  @AuthorizationConfiguration.RequirePermissionUpdateTestForTestOrder
+  public TestOrder editQueueItemMultiplex(
+      UUID testOrderId,
+      UUID deviceSpecimenTypeId,
+      List<MultiplexTestResult> results,
+      Date dateTested) {
+    TestOrder savedOrder =
+        editQueueItem(
+            testOrderId,
+            deviceSpecimenTypeId,
+            getCovidResultFromMultiplexList(results).toString(),
+            dateTested);
+
+    editMultiplexResults(savedOrder, results);
+
+    return savedOrder;
+  }
+
   @AuthorizationConfiguration.RequirePermissionSubmitTestForPatient
   @Transactional(noRollbackFor = {TwilioException.class, ApiException.class})
   public AddTestResultResponse addTestResult(
       UUID deviceSpecimenTypeId, TestResult result, UUID patientId, Date dateTested) {
+    TestOrder savedOrder =
+        saveTestResultToDatabase(deviceSpecimenTypeId, result, patientId, dateTested);
+
+    ArrayList<Boolean> deliveryStatuses = new ArrayList<>();
+
+    PatientLink patientLink = _pls.createPatientLink(savedOrder.getInternalId());
+    if (patientHasDeliveryPreference(savedOrder)) {
+
+      if (smsDeliveryPreference(savedOrder) || smsAndEmailDeliveryPreference(savedOrder)) {
+        boolean smsDeliveryStatus = testResultsDeliveryService.smsTestResults(patientLink);
+        deliveryStatuses.add(smsDeliveryStatus);
+      }
+
+      if (emailDeliveryPreference(savedOrder) || smsAndEmailDeliveryPreference(savedOrder)) {
+        boolean emailDeliveryStatus = testResultsDeliveryService.emailTestResults(patientLink);
+        deliveryStatuses.add(emailDeliveryStatus);
+      }
+    }
+
+    boolean deliveryStatus =
+        deliveryStatuses.isEmpty() || deliveryStatuses.stream().anyMatch(status -> status);
+    return new AddTestResultResponse(savedOrder, deliveryStatus);
+  }
+
+  @AuthorizationConfiguration.RequirePermissionSubmitTestForPatient
+  public AddTestResultResponse addTestResultMultiplex(
+      UUID deviceSpecimenTypeId,
+      List<MultiplexTestResult> results,
+      UUID patientId,
+      Date dateTested) {
+    AddTestResultResponse response =
+        addTestResult(
+            deviceSpecimenTypeId, getCovidResultFromMultiplexList(results), patientId, dateTested);
+    TestOrder savedOrder = response.getTestResult().getWrapped();
+    TestEvent savedEvent = savedOrder.getTestEvent();
+
+    saveMultiplexResults(savedEvent, savedOrder, results);
+
+    return response;
+  }
+
+  @AuthorizationConfiguration.RequirePermissionSubmitTestForPatient
+  @Transactional(noRollbackFor = {TwilioException.class, ApiException.class})
+  public TestOrder saveTestResultToDatabase(
```

Comment:
nit: given that this method is only used in one place, should we keep it?

I think I originally broke this out because I didn't think the results were saving correctly, but they're able to save now without the additional `@Transactional` method.

Reply:
Good point - I think I had been calling this in `addTestResultMultiplex` at one point, but then I just changed that method to call `addTestResult`, so I think this is no longer needed.

Reply:
Doing this made it count as new code, and my test coverage dropped to 27% 😩 though I'm a little confused because it should be covered already by existing tests? Looking into it

Reply:
Sonar is claiming that TestOrderService is completely uncovered now 🤔